dataclass_csv/decorators.py

```diff
@@ -1,4 +1,8 @@
-def dateformat(date_format):
+from typing import Any, Callable, TypeVar, Type
+
+F = TypeVar('F', bound=Callable[..., Any])
```

Comment:
Super cool! 🙌🏼

Reply:
This is the standard method of doing decorators lifted from mypy's docs. ;) "cool" is one way to describe it. I kinda just stare at it in wonder and awe. ;)